packages/x-data-grid/src/components/columnSelection/GridCellCheckboxRenderer.tsx

```diff
@@ -50,6 +52,9 @@ const GridCellCheckboxForwardRef = React.forwardRef<HTMLInputElement, GridRender
     const classes = useUtilityClasses(ownerState);
     const checkboxElement = React.useRef<HTMLElement>(null);
 
+    const someChildrenSelectedSelector = getGridSomeChildrenSelectedSelector(id);
+    const someChildrenSelected = useGridSelector(apiRef, someChildrenSelectedSelector);
```

Comment:
This is essentially selectors with external arguments. I've been wondering how to do this, it would have been nice to be able to do `useGridSelector(apiRef, selector, arg1, arg2, ...)`, but I think the 3rd parameter is already used for something.

Reply:
Yes. The second argument of `OutputSelector` and third argument of `useGridSelector` are used for something else. We would need to make breaking changes to `createSelector` and `useGridSelector`. How about we plan it for v8?

Or would it make sense to introduce a v8-only API now while deprecating the current one?

Reply:
If you have the time to add a better API now I'd do that, I think it's going to feel cleaner to add selector arguments than selector creators. It could also wait for v8.

Reply:
We could do `useGridSelector(apiRef, selector, args, compareFunction)` to allow external arguments, though I'd prefer to do that separately because only allowing the selector arguments won't solve all the needs for this particular use case.
For example, we need `apiRef` access in the `gridRowGroupSelectableChildrenSelector` selector to use the `isRowSelectable` API method.

Reply:
Just realized, `apiRef` could be another argument 🙂. [Added](https://github.com/mui/mui-x/pull/13757/commits/8e5238ecfcacf2489899426aa46c19aa36c196ae) an initial implementation for the selectors with arguments.

**Selector Creation:**
```tsx
const someSelector = createSelectorV8(
  selector1,
  selector2,
  ...,
  selectorN,
  (result1, result2, ..., resultN, args) => { 
    // logic for output function 
  })
);
```

**Usage:**
```tsx
const args = {
  arg1: 'arg1',
  arg2: 'arg2',
}
const value = useGridSelectorV8(apiRef, selector, args, compareFunction)
```

Still figuring out how to type the arguments though.
Any leads or comments on the API would be appreciated.

Reply:
Extracted to https://github.com/mui/mui-x/pull/14236
Will remove the diff from here.